Index: src/main/java/org/example/Directorio.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/org/example/Directorio.java b/src/main/java/org/example/Directorio.java
new file mode 100644
--- /dev/null	(date 1683824801527)
+++ b/src/main/java/org/example/Directorio.java	(date 1683824801527)
@@ -0,0 +1,31 @@
+package org.example;
+
+import java.util.ArrayList;
+import java.util.List;
+public class Directorio implements Componente {
+    private String nombre;
+    private List<Componente> componentes;
+
+    public Directorio(String nombre) {
+        this.nombre = nombre;
+        this.componentes = new ArrayList<>();
+    }
+
+    @Override
+    public String getNombre() {
+        return nombre;
+    }
+
+    @Override
+    public int getTamano() {
+        int total = 0;
+        for (Componente componente : componentes) {
+            total += componente.getTamano();
+        }
+        return total;
+    }
+
+    public void agregarComponente(Componente componente) {
+        componentes.add(componente);
+    }
+}
